port/resource_port_blueprint.go

```diff
@@ -0,0 +1,312 @@
+							Optional:    true,
+							Description: "Whether or not the relation is required",
+						},
+						"many": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Whether or not the relation is many",
+						},
+					},
+				},
+				Optional: true,
+			},
+			"properties": {
+				Description: "The metadata of the entity",
+				Type:        schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"identifier": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The identifier of the property",
+						},
+						"title": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The name of this property",
+						},
+						"type": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The type of the property",
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The description of the property",
+						},
+						"default": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The default value of the property",
+						},
+						"format": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The format of the Property",
+						},
+					},
+				},
+				Required: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func readBlueprint(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	c := m.(*cli.PortClient)
+	b, err := c.ReadBlueprint(ctx, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	writeBlueprintFieldsToResource(d, b)
+	relations, err := c.ReadRelations(ctx, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	writeBlueprintRelationsToResource(d, relations)
+	return diags
+}
+
+func writeBlueprintRelationsToResource(d *schema.ResourceData, relations []*cli.Relation) {
+	rels := schema.Set{F: func(i interface{}) int {
+		id := (i.(map[string]interface{}))["identifier"].(string)
+		return schema.HashString(id)
+	}}
+	for _, v := range relations {
+		r := map[string]interface{}{
+			"identifier": v.Identifier,
+			"title":      v.Title,
+			"target":     v.Target,
+			"required":   v.Required,
+			"many":       v.Many,
+		}
+		rels.Add(r)
+	}
+	d.Set("relations", &rels)
+}
+
+func writeBlueprintFieldsToResource(d *schema.ResourceData, b *cli.Blueprint) {
+	d.SetId(b.Identifier)
+	d.Set("title", b.Title)
+	d.Set("icon", b.Icon)
+	d.Set("data_source", b.DataSource)
+	d.Set("created_at", b.CreatedAt.String())
+	d.Set("created_by", b.CreatedBy)
```

Comment:
CreatedBy should be also in string?

Reply:
no, only `time.Time` should be printed with `.String()` (`CreatedBy` is of type string)